ariadne_graphql_modules/next/interfacetype.py

```diff
@@ -0,0 +1,220 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Type,
+    Union,
+    cast,
+    Sequence,
+)
+
+from ariadne import InterfaceType
+from ariadne.types import Resolver
+from graphql import (
+    FieldDefinitionNode,
+    GraphQLField,
+    GraphQLObjectType,
+    GraphQLSchema,
+    InputValueDefinitionNode,
+    NameNode,
+    NamedTypeNode,
+    InterfaceTypeDefinitionNode,
+)
+
+from .value import get_value_node
+
+from .objecttype import (
+    GraphQLObject,
+    GraphQLObjectResolver,
+    get_field_args_from_resolver,
+    get_field_args_out_names,
+    get_field_node_from_obj_field,
+    get_graphql_object_data,
+    update_field_args_options,
+)
+
+from ..utils import parse_definition
+from .base import GraphQLMetadata, GraphQLModel, GraphQLType
+from .description import get_description_node
+
+
+class GraphQLInterface(GraphQLObject):
+    __types__: Sequence[Type[GraphQLType]]
+    __implements__: Optional[Iterable[Union[Type[GraphQLType], Type[Enum]]]]
```

Comment:
`__interface__` is mixed with `__types__`. Types can't implement enums, but can have dependency on them.